https://github.com/pulumi/pulumi/actions/runs/26098947344/job/76744830379?pr=23272
```
Run make -C npm test
make: *** npm: No such file or directory.  Stop.
Error: Process completed with exit code 2.
```

Does this just need a rebase?

--- a/sdk/go/common/util/cmdutil/term.go
+++ b/sdk/go/common/util/cmdutil/term.go
@@ -101,11 +101,10 @@ func TerminateProcessGroup(proc *os.Process, cooldown time.Duration) (ok bool, e
 		return false, killProcessGroup(proc)
 	}
 
-	var waitErr error
+	waitErrCh := make(chan error, 1)
 	ctx, cancel := context.WithTimeout(context.Background(), cooldown)
+	defer cancel()
 	go func() {
-		defer cancel()
-
 		state, err := proc.Wait()
 		switch {
 		case err == nil && !state.Success():
@@ -116,7 +115,8 @@ func TerminateProcessGroup(proc *os.Process, cooldown time.Duration) (ok bool, e
 			err = nil
 		}
 
-		waitErr = err
+		waitErrCh <- err
+		cancel()
 	}()
 
 	// The context will be canceled when the timeout expires,
@@ -129,5 +129,5 @@ func TerminateProcessGroup(proc *os.Process, cooldown time.Duration) (ok bool, e
 		return false, killProcessGroup(proc)
 	}
 
-	return true, waitErr
+	return true, <-waitErrCh
 }

